source/Firmata/UwpFirmata.cpp

```diff
@@ -234,7 +238,156 @@ UwpFirmata::processInput(
+
+        vector.push_back( static_cast<uint8_t>( data & 0xFF ) );
+        ++bytes_read;
+        --bytes_remaining;
+    }
+
+    //process the message
+    switch( command )
+    {
+        //ignore these message types
+    default:
+    case Command::REPORT_ANALOG_PIN:
+    case Command::REPORT_DIGITAL_PIN:
+    case Command::SET_PIN_MODE:
+    case Command::END_SYSEX:
+    case Command::SYSTEM_RESET:
+    case Command::PROTOCOL_VERSION:
+        return;
+
+    case Command::ANALOG_MESSAGE:
+        //report analog commands store the pin number in the lower nibble of the command byte, the value is split over two 7-bit bytes
+        AnalogValueUpdated( this, ref new CallbackEventArgs( lower_nibble, vector.at( 0 ) | ( vector.at( 1 ) << 7 ) ) );
+        break;
+
+    case Command::DIGITAL_MESSAGE:
+        //digital messages store the port number in the lower nibble of the command byte, the port value is split over two 7-bit bytes
+        DigitalPortValueUpdated( this, ref new CallbackEventArgs( lower_nibble, vector.at( 0 ) | ( vector.at( 1 ) << 7 ) ) );
+        break;
+
+    case Command::START_SYSEX:
+        //a sysex message must include at least one extended-command byte
+        if( bytes_read < 1 ) return;
+
+        //retrieve the raw data array & extract the extended-command byte
+        uint8_t *arr = vector.data();
+        SysexCommand sysCommand = static_cast<SysexCommand>( arr[0] );
+        ++arr;
+        --bytes_read;
+
+        DataWriter ^writer = ref new DataWriter();
+        switch( sysCommand )
+        {
+        case SysexCommand::STRING_DATA:
+
+            //condense back into 1-byte data
+            reassembleByteString( arr, bytes_read );
+
+            StringMessageReceived( this, ref new StringCallbackEventArgs( createStringFromMbs( arr, bytes_read / 2 ) ) );
+
+        break;
+
+        case SysexCommand::CAPABILITY_RESPONSE:
+
+            //Firmata does not handle capability responses in the typical way (separating bytes), so we write them directly to the DataWriter
+            for( int i = 0; i < bytes_read; ++i )
+            {
+                writer->WriteByte( arr[i] );
+            }
+            PinCapabilityResponseReceived( this, ref new SysexCallbackEventArgs( static_cast<uint8_t>( sysCommand ), writer->DetachBuffer() ) );
+
+            break;
+
+        case SysexCommand::I2C_REPLY:
+
+            //condense back into 1-byte data
+            reassembleByteString( arr, bytes_read );
+
+            //if we're receiving an I2C reply, the first two bytes in our reply are the address and register
+            for( int i = 2; i < bytes_read / 2; ++i )
+            {
+                writer->WriteByte( arr[i] );
+            }
+
+            I2cReplyReceived( this, ref new I2cCallbackEventArgs( arr[0], arr[1], writer->DetachBuffer() ) );
+            break;
+
+        default:
+
+            //condense back into 1-byte data
+            reassembleByteString( arr, bytes_read );
```

Comment:
I don't think all sysEx messages conform to always having every value split into msb/lsb pairs.  It might be better to have the caller parse the sysex message as appropriate.